content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,329 @@
+---
+title: 'Referential actions'
+metaTitle: 'Referential actions'
+metaDescription: 'Referential actions let you define the update and delete behavior of related models on the database level'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+As of [2.26.0](https://github.com/prisma/prisma/releases/tag/2.26.0] Prisma now supports _referential actions_ as a Preview feature part of the Prisma Schema Language.
+
+This is a [**preview feature**](../../../about/releases#preview) and can be accessed by:
+
+1. [Upgrading to version v2.26.0 of Prisma](../../../guides/upgrade-guides/upgrading-to-latest).
+2. Adding `referentialActions` to the `previewFeatures` field of the Prisma Client [`generator`](../../../reference/api-reference/prisma-schema-reference#generator) block in the `prisma.schema` file.
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialActions"]
+}
+```
+
+After enabling the preview feature flag you now have 3 options:
+
+1. Make your schema reflect the current state of the database by running [Introspection](../introspection). Afterwards you can see the referential actions of all relations your relations explicitly, which might be different than the new defaults.
+2. Define custom referential actions on your data model relation fields. You then update the database schema by using [Prisma Migrate](../prisma-migrate) or the [`db push`](../prisma-migrate/db-push) command.
+3. Apply the new default referential actions, that exist when no custom actions are added manually, by using [Prisma Migrate](../prisma-migrate), or [`db push`](../prisma-migrate/db-push), without making any changes to your existing Prisma schema.
+
+To begin using referential actions you add them to the `@relation` attribute in your schema.
+
+```prisma
+@relation(fields: [authorId], references: [id], onDelete: Cascade)
+```
+
+See [Upgrade guides](#upgrade-guides) for more information on adopting the feature in your project.
+
+</TopBlock>
+
+## What are referential actions?
+
+Referential actions are constraints that can be used to define how a referenced record is handled by the database when either an [`update`](../prisma-client/crud#update) or [`delete`](../prisma-client/crud#delete) query is run.
+
+These include:
```

Comment:
Include suggestions that that isn't a complete list - maybe "Prisma supports the following referential actions:"